Pull Request for updating stm.py to provide top 5 heat maps

CodeMaster001 opened the pull request:
1.Placed stm.py under the src directory
2.Renamed freq to top for better understanding.

CodeMaster001 committed:
Updated clia stm for top 5 heatmaps

CodeMaster001 committed:
Merge pull request #1 from CodeMaster001/CodeMaster001-patch-1

Updated clia stm for top 5 heatmaps

CodeMaster001 committed:
Put stm.py inside source directory

Changed stm.py file location

CodeMaster001 committed:
Delete stm.py outside source

CodeMaster001 committed:
Renames freq to top

CodeMaster001 committed:
Merge pull request #2 from CodeMaster001/CodeMaster001-patch-1

Code master001 patch 1

CodeMaster001 committed:
Add files via upload

CodeMaster001 committed:
Updated stm

-Fixed Styling

magsol commented on src/stm.py:
```diff
@@ -4,7 +4,7 @@
 # Main function
 
-def cbf(volume, method = "fft", fps = 200, max_freq = 20):
+def cbf(volume, method = "fft", fps = 200, max_freq = 20, sequence = 5):
```
Rename `sequence` to `n_maps`. Default of 5 is fine.

magsol commented on src/stm.py:
```diff
@@ -30,18 +30,18 @@ def cbf(volume, method = "fft", fps = 200, max_freq = 20):
     """
     if method == "fft":
-        return _cbf_fft(volume, fps, max_freq = max_freq)
+        return _cbf_fft(volume, fps, max_freq = max_freq,sequence=sequence)
```
Add proper spacing in accordance with PEP standards:
 - there should be a space after a comma
 - consistent with the rest of the file, there should be spaces on either sides of the equals `=` sign

magsol commented on src/stm.py:
```diff
@@ -30,18 +30,18 @@ def cbf(volume, method = "fft", fps = 200, max_freq = 20):
+        return _cbf_fft(volume, fps, max_freq = max_freq,sequence=sequence)
     elif method == "psd":
-        return _cbf_psd(volume, fps, max_freq = max_freq)
+        return _cbf_psd(volume, fps, max_freq = max_freq,sequence=sequence)
```
Same as above.

magsol commented on src/stm.py:
```diff
@@ -30,18 +30,18 @@ def cbf(volume, method = "fft", fps = 200, max_freq = 20):
+        return _cbf_psd(volume, fps, max_freq = max_freq,sequence=sequence)
     elif method == "welch":
-        return _cbf_welch(volume, fps, max_freq = max_freq)
+        return _cbf_welch(volume, fps, max_freq = max_freq,sequence=sequence)
```
Same as above.

magsol commented on src/stm.py:
```diff
@@ -30,18 +30,18 @@ def cbf(volume, method = "fft", fps = 200, max_freq = 20):
 # CBF functions
 
-def _cbf_fft(volume, fps, max_freq = 20):
+def _cbf_fft(volume, fps, max_freq = 20, sequence = 5):
```
Same as above (rename `sequence` to `n_maps`)

magsol commented on src/stm.py:
```diff
@@ -107,12 +97,9 @@ def _cbf_psd(volume, fps, max_freq = 20):
+    return get_best_volumes(Pxx, f, max_freq, sequence)
 
-def _cbf_welch(volume, fps, max_freq = 20):
+def _cbf_welch(volume, fps, max_freq = 10, sequence = 5):
```
Keep `max_freq` default value the same across all functions (others are all 20, not 10).

magsol commented on src/stm.py:
```diff
@@ -136,13 +123,26 @@ def _cbf_welch(volume, fps, max_freq = 20):
-    heatmap = f[max_freq_indices]
-    heatmap[heatmap > max_freq] = max_freq
-    return heatmap
+    return get_best_volumes(Pxx,f,max_freq,sequence)
```
Spacing after commas.

magsol replied:
Still need to fix the spacing here in the arglist.

magsol commented on src/stm.py:
```diff
@@ -136,13 +123,26 @@ def _cbf_welch(volume, fps, max_freq = 20):
+        heatmap[heatmap > max_freq] = max_freq
+        heatmap_list.append(heatmap)
+    return np.array(heatmap_list)
+
```
Spurious newlines between end of this function and header of next function. Again, per PEP guidelines, should only be 1 blank line between function definitions.

magsol commented on src/stm.py:
```diff
@@ -136,13 +123,26 @@ def _cbf_welch(volume, fps, max_freq = 20):
 
 # Utilities
 
+def get_best_volumes(pxx, f, max_freq, top = 5):
```
Several items:
 - rename function to `_get_best_volumes`, given its role as a utility that is only called by other utilities.
 - rename `top` to `n_maps`, again consistent with previous naming conventions
 - rename `pxx` and `f` to something more meaningful to the user (e.g., `pow_spec` and `freq_bins`, respectively).

magsol commented on src/stm.py:
```diff
@@ -136,13 +123,26 @@ def _cbf_welch(volume, fps, max_freq = 20):
 # Utilities
 
+def get_best_volumes(pxx, f, max_freq, top = 5):
+    max_freq_indices = np.array(- pxx).argsort(axis = 0)[:top]
```
`pxx` is already a NumPy array; shorten to simply `(-pxx).argsort(axis = 0)`, etc

magsol reviewed (changes requested):
Numerous changes requested. More to come after this first round has been addressed.

Commit:
Delete stm.py

CodeMaster001 committed:
Added stm.py

CodeMaster001 committed:
Fixed stm.py

magsol commented on src/stm.py:
```diff
@@ -1,14 +1,14 @@
     """
     Convenience CBF function that invokes one of the three methods for
     computing CBF used in Quinn et al 2015, Science Translational Medicine.
-
```
This blank line _should_ be here.

magsol commented on src/stm.py:
```diff
@@ -23,30 +23,28 @@ def cbf(volume, method = "fft", fps = 200, max_freq = 20):
         Framerate of the video, or frames per second (default: 200).
     max_freq : float
         Maximum allowable frequency; anything above this is clamped (default: 20).
-
```
Add this blank line back in.

magsol commented on src/stm.py:
```diff
@@ -23,30 +23,28 @@ def cbf(volume, method = "fft", fps = 200, max_freq = 20):
     Calculates the ciliary beat frequency (CBF) of a given 3D volume. This
     function is fast but produces noisy results, as it only considers the
     instantaneous signal when extracting frequencies.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -55,7 +53,6 @@ def _cbf_fft(volume, fps, max_freq = 20):
         Frames per second (capture framerate) of the video.
     max_freq : float
         Maximum allowed frequency; frequencies above this are clamped.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -73,23 +70,12 @@ def _cbf_fft(volume, fps, max_freq = 20):
     """
     Calculates the ciliary beat frequency (CBF) of a given 3D volume. This
     function computes a simple periodogram of frequencies in a given signal.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -98,7 +84,6 @@ def _cbf_psd(volume, fps, max_freq = 20):
         Frames per second (capture framerate) of the video.
     max_freq : float
         Maximum allowed frequency; frequencies above this are clamped.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -107,18 +92,14 @@ def _cbf_psd(volume, fps, max_freq = 20):
     function uses the Welch algorithm to build a periodogram that is smoothed
     using a windowed average, generating a robust estimation of the spectral
     density of the signal.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -127,31 +108,49 @@ def _cbf_welch(volume, fps, max_freq = 20):
         Frames per second (capture framerate) of the video.
     max_freq : float
         Maximum allowed frequency; frequencies above this are clamped.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -127,31 +108,49 @@ def _cbf_welch(volume, fps, max_freq = 20):
 def nextpow2(i):
     """
     Calculates the next even power of 2 which is greater than or equal to i.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -127,31 +108,49 @@ def _cbf_welch(volume, fps, max_freq = 20):
     -----------
     i : integer
         The number which we want a power of 2 that is greater.
-
```
Add blank line back in.

magsol commented on src/stm.py:
```diff
@@ -160,4 +159,4 @@ def nextpow2(i):
     while n < i:
         n *= 2
-    return n
+    return n
```
Add 1 blank line at end of file.

magsol commented on src/stm.py:
```diff
@@ -127,31 +108,49 @@ def _cbf_welch(volume, fps, max_freq = 20):
 
 # Utilities
 
+def _get_best_volumes(freq_bins, f, max_freq = 20, n_maps = 5):
```
This function header has a leading underscore in the name, but none of the calls to this function have been updated to reflect that change. In short, this code won't work. Please fix in the function calls.

magsol commented on src/stm.py:
```diff
@@ -127,31 +108,49 @@ def _cbf_welch(volume, fps, max_freq = 20):
+def _get_best_volumes(freq_bins, f, max_freq = 20, n_maps = 5):
+    """
+    The Utility function gets the best maps the top 20 maps from the function
+    Parameters
```
Add a space between the function doc and the `Parameters` header.

magsol commented on src/stm.py:
```diff
@@ -127,31 +108,49 @@ def _cbf_welch(volume, fps, max_freq = 20):
+        Maximum allowed frequency; frequencies above this are clamped.
+    n_maps : float
+        top n maps to be selected for heatmaps
+    Returns
```
Add a space between the end of the parameters description and the `Returns` header.

magsol reviewed (changes requested):
More changes need to be addressed.

CodeMaster001 committed:
updated stm.py

CodeMaster001 committed:
Updated stm for code fix

Updated stm.py file for code fix

CodeMaster001 committed:
Delete stm.py

CodeMaster001 committed:
Updating stm.py

-Added stm.py file

CodeMaster001 committed:
Added stm.py

magsol commented:
Code style changes have still not been addressed. Furthermore, a duplicate file `stm.py` was added to the root, which appears to be a duplicate of the original, but may not be, and will only introduce additional complications. Finally, commit messages have been uninformative ("updated stm.py", "delete stm.py", "added stm.py"), further complicating my ability to review changes. Unless the requested changes are implemented and good coding practices are adhered to, I am going to close this PR.

Commit:
Updated stm.py
1.Added spaces after main funciton
2.Added space before the actual function code
3.Added space between parameters
